transport/TransportFactory.cpp

```diff
@@ -484,6 +484,44 @@ class TransportFactoryImpl final : public TransportFactory,
         endpoint->stop(this);
     }
 
+    void registerIceListener(Endpoint::IEvents& listener, const std::string& ufrag) override
+    {
+        for (auto& endpoints : _sharedEndpoints)
+        {
+            for (auto& endpoint : endpoints)
```

Comment:
I pondered this a bit. Each vector in _sharedEndpoints contains all endpoints for all interfaces but with a specific port eg. 10000. The next vector will be the same set of interfaces but port 10001 for example. 
There is no point in client trying to reach all interfaces multiple times. It is ok if the probes are all on the first port (10000). The multi-port is to spread load from all RTP. For ICE probes it does not matter.
Once this test probe is done, the client will restart ICE anyway on the port selected for join.
Ie. do not loop over _sharedEndpoints, take the first one if it exists.

Reply:
Yes, I think we discussed it with @RicardoMDomingues before as initially I only took first UDP endpoint. The concern was that all the probes will be sent to single port. But I can change it back to one port.

Reply:
Yes, I mean we have discussed about opening a range of ports in smb because we have already detected our send limit in a single socket is not that high.
Opening different ports on same interface should generate candidates with same foundation. If CS see different candidates with same foundation, it can round robin the candidates provided to client and even provide 2 just to reduce the probability of some glitch caused by an overloaded socket, @olofkallander, wdyt?

Reply:
Changed to pass endpoints for 1st port only.